dvc/repo/experiments/executor/base.py

```diff
@@ -450,9 +455,27 @@ def _repro_args(cls, dvc):
             kwargs = {}
         return args, kwargs
 
+    @staticmethod
+    def _auto_push(git_remote: str, dvc: "Repo", scm: "Git"):
+        if git_remote == scm.root_dir:
+            logger.warning(
+                f"try to auto checkpoints to {git_remote} which is the "
+                "running repository dvc cache will be pushed to the "
+                "default remote while git references will not be pushed"
```

Comment:
I'm not sure I understand this message. Maybe something like

> Try to save the experiment status to {git_remote} after every checkpoint.
Data will be pushed to the default remote.

Not sure what Git refecences we're talking about. Meaning the experiments themselves? If so what gets saved to {git_remote} ?

Reply:
We are `git push`ing the currently running experiment ref. In this condition, `git_remote` points to the current git repo itself, so nothing actually happens on the git side. However, when this env var is set, we also automate `dvc push`ing used cache objects and run cache for the currently running experiment as well.

So the side effect here is that nothing will be "pushed" anywhere on the git side of things, but DVC outputs and run cache will still be pushed to the default configured DVC remote.

Reply:
The warning message should probably just be something like

> '{git_remote}' points to the current Git repo, experiment Git refs will not be pushed. DVC cache and run cache will automatically be pushed to the default DVC remote (if any) on each experiment commit.